code/go/0chain.net/smartcontract/zcnsc/authorizer.go

```diff
@@ -53,24 +54,15 @@ func (zcn *ZCNSmartContract) AddAuthorizer(
 		Logger.Error("public key error", zap.Error(err))
 		return "", err
 	}
-
-	if params.ID == "" {
-		err = common.NewError(code, "authorizer ID cannot be empty")
-		Logger.Error("authorizer id error", zap.Error(err))
-		return "", err
-	}
-
-	authorizerID = params.ID
-
 	if params.PublicKey == "" {
 		err = common.NewError(code, "public key was not included with transaction")
 		Logger.Error("public key error", zap.Error(err))
 		return "", err
 	}
 
 	publicKeyBytes, _ := hex.DecodeString(params.PublicKey)
```

Comment:
As commented in another PR, let's not ignore the decode error.

Reply:
@peterlimg usually, we don't improve code we haven't touched in the review, it can slowdown the process severely

Reply:
@uneet7 I've fixed it, but please handle all errors you get especially in smart contracts

Reply:
@dabasov it's a minor fix that does not worth a new PR, otherwise we might just ignore it and forget it. The process would only be slowdown when we let code go to production easily and then spend more time to debug/fix it later .